x-pack/plugins/fleet/server/services/agents/action_status.ts

```diff
@@ -42,13 +42,40 @@ export async function getActionStatuses(esClient: ElasticsearchClient): Promise<
 
       const nbAgentsActioned = action.nbAgentsActioned || action.nbAgentsActionCreated;
       const complete = count === nbAgentsActioned;
-      const isCancelled = cancelledActionIds.indexOf(action.actionId) > -1;
+      const cancelledAction = cancelledActions.find((a) => a.actionId === action.actionId);
+      let completionTime: string | undefined;
+
+      if (complete) {
+        // querying the last action result for this action in case all agents acked, to report completion time
+        const { aggregations } = await esClient.search({
```

Comment:
I didn't find a better way to determine the completion time of an action, other than to query the last timestamp of action results for this action. Maybe we could have a field on the action document itself to record the completion time when the last agent acked the action.